src/main/java/org/spongepowered/api/text/conversation/QuestionResult.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.conversation;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+/**
+ * What to return from your answer handler on a question, to allow moving on to
+ * a next question, the same question, or to end the conversation.
+ */
+public class QuestionResult {
```

Comment:
Most of this is implementation that should be in Common.

That said, I can only see that there are _two_ options for returning a result, ask a question, or end the conversation. Is there value in having a `REPEAT` type - instead just setting the question again (which you already have from the `AnswerHandler`)? If not, then I suggest stripping the types out.

I understand having this result object (though maybe as an interface), in case we want to add more information in the future should the need arise. For example, this could probably extend (or at least use a subset of) `CommandResult` though the usefulness of that is probably debatable.